test/Benchmarks/src/Vector/Operations.enso

```diff
@@ -16,31 +16,40 @@ num_iterations = 10
-    Bench.measure (random_vec.reduce (+)) "Sum" iter_size num_iterations
-    Bench.measure ((random_vec.drop (First 20)).reduce (+)) "Drop First 20 and Sum" iter_size num_iterations
-    Bench.measure ((random_vec.drop (Last 20)).reduce (+)) "Drop Last 20 and Sum" iter_size num_iterations
-    Bench.measure (random_vec.filter (x -> x % 3 == 1)) "Filter" iter_size num_iterations
-    Bench.measure (random_vec.filter_with_index (i-> x-> (i+x) % 3 == 1)) "Filter With Index" iter_size num_iterations
-
-    Bench.measure (random_vec . map (x -> x +  random_gen.nextLong) . filter (x -> x % 3 == 1)) "Map & Filter" iter_size num_iterations
-    Bench.measure (random_vec.partition (x -> x % 3 == 1)) "Partition" iter_size num_iterations
-    Bench.measure (random_vec.partition_with_index (i-> x-> (i+x) % 3 == 1)) "Partition With Index" iter_size num_iterations
+    bench_measure ~act name _ _ = group_builder.specify name act
+
+    bench_measure (Base.Vector.new vector_size i->i) "New Vector" iter_size num_iterations
+    bench_measure (Base.Vector.new vector_size _->42) "New Constant" iter_size num_iterations
+    bench_measure (Base.Vector.new vector_size _->random_gen.nextLong) "New Random" iter_size num_iterations
+    bench_measure (Base.Vector.fill vector_size 42) "Fill Constant" iter_size num_iterations
+    bench_measure (Base.Vector.fill vector_size random_gen.nextLong) "Fill Random (constant)" iter_size num_iterations
+    bench_measure (random_vec + [1]) "Append Single" iter_size num_iterations
+    bench_measure (random_vec + random_vec_2) "Append Large" iter_size num_iterations
+    bench_measure (random_vec.reduce (+)) "Sum" iter_size num_iterations
+    bench_measure ((random_vec.drop (First 20)).reduce (+)) "Drop First 20 and Sum" iter_size num_iterations
+    bench_measure ((random_vec.drop (Last 20)).reduce (+)) "Drop Last 20 and Sum" iter_size num_iterations
+    bench_measure (random_vec.filter (x -> x % 3 == 1)) "Filter" iter_size num_iterations
+    bench_measure (random_vec.filter_with_index (i-> x-> (i+x) % 3 == 1)) "Filter With Index" iter_size num_iterations
+
+    bench_measure (random_vec . map (x -> x +  random_gen.nextLong) . filter (x -> x % 3 == 1)) "Map & Filter" iter_size num_iterations
+    bench_measure (random_vec.partition (x -> x % 3 == 1)) "Partition" iter_size num_iterations
+    bench_measure (random_vec.partition_with_index (i-> x-> (i+x) % 3 == 1)) "Partition With Index" iter_size num_iterations
 
     stateful_fun x =
        s = State.get Number
        State.put s+x
-    Bench.measure (State.run Number 0 <| random_vec.each stateful_fun) "Each" iter_size num_iterations
+    bench_measure (State.run Number 0 <| random_vec.each stateful_fun) "Each" iter_size num_iterations
 
-main = bench
+main =
+    options = Bench.options . size iter_size . iter num_iterations
+
+    all = Bench.build builder->
+        builder.group "Vector Operations" options group_builder->
+            bench group_builder
+
```

Comment:
@Akirathan, my idea for integration with JMH is to change all the benchmarks like I did in [this commit](https://github.com/enso-org/enso/pull/7101/commits/88fd6fb9888d939c2d9745a7397f6e87dec4fa4d). The JMH then needs to get access to `all` and _not call_ `run_main` on them, but iterate over the benchmarks to find the one with right name and execute it in JMH.

If we agree that changing the benchmark registration to the builder pattern is a way to go, then you can prepare a generic JMH benchmark that would cooperate with `Bench`, `Bench_Options` & co.